lib/flutter_accessorysetup.dart

```diff
@@ -1,20 +1,222 @@
-import 'package:flutter_accessorysetup/flutter_accessorysetup_event.dart';
-import 'package:flutter_accessorysetup/flutter_accessorysetup_platform.dart';
+import 'dart:async';
+import 'dart:io';
 
-class FlutterAccessorysetup {
-  FlutterAccessorysetupPlatform get _platform => FlutterAccessorysetupPlatform.instance;
+import 'package:objective_c/objective_c.dart';
+import 'package:flutter/services.dart';
 
-  Stream<FlutterAccessorysetupSessionEvent> get sessionStream => _platform.sessionStream;
+import 'package:flutter_accessorysetup/gen/ios/accessory_setup_bindings.dart';
+import 'package:flutter_accessorysetup/helpers.dart';
 
-  Future<String?> getPlatformVersion() {
-    return _platform.getPlatformVersion();
+class FlutterAccessorysetupFFI {
+  Stream<ASAccessoryEvent> get eventStream => _eventsController.stream;
+  List<ASAccessory> get accessories => _session.accessories.toList();
+  final _session = FFIAccessorySession.alloc().init();
+  final _eventsController = StreamController<ASAccessoryEvent>();
+
+  late final _delegate = FFIAccessorySessionDelegate.implementAsListener(
+      handleEvent_: _handleEvent,
+      didShowPickerWithError_: _didShowPicker,
+      didRenameAccessory_withError_: _didRenameAccessory,
+      didRemoveAccessory_withError_: _didRemoveAccessory,
+      didFinishAuthorizationForAccessory_withError_: _didFinishAuthorization,
+      didFailAuthorizationForAccessory_withError_: _didFailAuthorization);
+  // TODO: bind completers to accessories to enable multiple calls
```

Comment:
is this still pending?

Reply:
yes, it is a todo. Might be worth in the future